### PR TITLE
Fluent UI migration Model Page

### PR DESCRIPTION
**Pull request by RHeckerIntel**
Implements the model page:
Screenshot 2024-11-13 at 13 32 45

Dark mode:
Screenshot 2024-11-13 at 13 35 02

### CODE REVIEW

File: lib/pages/models/widgets/model_list.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Padding(
+              padding: const EdgeInsets.only(left: 65, right: 65, top: 24),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  SearchBar(onChange: (f) => filter.name = f),
+                  IconButton(icon: const Icon(FluentIcons.sort), onPressed: () => filter.order = !filter.order),
```
Comment (ActiveChooN) [inline lib/pages/models/widgets/model_list.dart:27]:
```suggestion
                  IconButton(icon: const Icon(filter.order ? FluentIcons.ascending : FluentIcons.descending), onPressed: () => filter.order = !filter.order),
```

File: lib/pages/models/widgets/model_property.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 2),
+      child: Container(
+        decoration: BoxDecoration(
+          shape: BoxShape.rectangle,
+          borderRadius: const BorderRadius.all(Radius.circular(4)),
+          color: const Color(0xFFF5F5F5),
```
Comment (ActiveChooN) [inline lib/pages/models/widgets/model_property.dart:22]: Please use colors from theme if possible

File: lib/pages/models/widgets/model_card.dart [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+class _ModelCardState extends State<ModelCard> with SingleTickerProviderStateMixin {
+  bool isHovered = false;
+  late Animation<double> animation;
+  late AnimationController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(duration: const Duration(milliseconds: 300), vsync: this);
```
Comment (ActiveChooN) [inline lib/pages/models/widgets/model_card.dart:23]: Please use durations from the theme if possible

**Review by ActiveChooN — COMMENTED** [review]
LGTM, but have few minor comments. Also cards have overflow even on medium screen width
Screenshot 2024-11-08 at 12 55 46

File: lib/widgets/fixed_grid.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          final double sidePadding = (constraints.maxWidth - totalTilesWidth);
+
+          return Padding(
+            padding: EdgeInsets.only(right: sidePadding),
+            child: GridView.builder(
+              gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                crossAxisCount: columns,
+                crossAxisSpacing: spacing,
+                mainAxisSpacing: spacing,
+                childAspectRatio: 10/13
```
Comment (ActiveChooN) [inline lib/widgets/fixed_grid.dart:32]: Should this be available as parameter?

File: lib/pages/models/models.dart [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                            style: const TextStyle(
+                              fontSize: 20,
+                              fontWeight: FontWeight.w500,
+                            ),
+                          );
+                        }),
+                        FilledDropDownButton(
+                          title: const Text('Import model'),
+                          items: [
+                            MenuFlyoutItem(text: const Text('Hugging Face'), onPressed: () {}),
```
Comment (ActiveChooN) [inline lib/pages/models/models.dart:73]: Add a redirect to the import page since we have one, please

File: lib/pages/models/models.dart [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+              child: Column(
+                children: [
+                  GridContainer(
+                    color: theme.scaffoldBackgroundColor,
+                    padding: const EdgeInsets.all(13.5),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                      children: [
+                        Consumer<ProjectFilterProvider>(builder: (context, projectProvider, child) {
+                          return Text(projectProvider.option?.name ?? "",
```
Comment (ActiveChooN) [inline lib/pages/models/models.dart:63]: Do we need this header since option should be already highlighted in the side menu?

**Review by ActiveChooN — APPROVED** [review]
LGTM!

File: lib/widgets/fixed_grid.dart (hunk 4 above)
Comment (RHeckerIntel) [reply lib/widgets/fixed_grid.dart:32]: yes!

**Pull request merged**